webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -121,13 +124,15 @@ class ReportingServiceImpl(
 class CachedReportingServiceImpl(
     val defaultFindRuleNodeStatusReports: ReportingServiceImpl
   , val nodeInfoService                 : NodeInfoService
+  , val nodeConfigService               : NodeConfigurationService
   , val batchSize                       : Int
   , val complianceRepository            : ComplianceRepository
 ) extends ReportingService with RuleOrNodeReportingServiceImpl with CachedFindRuleNodeStatusReports {
   val confExpectedRepo = defaultFindRuleNodeStatusReports.confExpectedRepo
   val directivesRepo   = defaultFindRuleNodeStatusReports.directivesRepo
   val rulesRepo        = defaultFindRuleNodeStatusReports.rulesRepo
 
+  val nodeConfigrationService = nodeConfigService
```

Comment:
THIS is bothering me. i need to register the nodeConfigrationService because it in the CachedFindRuleNodeStatusReports

Reply:
perhaps you need to create a facade/new trait on top of `FindExpectedReportRepository` and `NodeConfigurationService` to only expose what is needed?

Reply:
Good idea !

Reply:
actually, it's never used, so I removed it